File: test-files/golden-tests/extensions/lua-set-name/addons/extensions/rename.lua
-- Mutate every function from a Lua extension. Exercises two shapes
-- of direct assignment: a scalar string write (`name`) and a
-- nested-object write whose leaves are `Polymorphic<Inline>` values
-- selected by a kebab-case `kind` tag.
--
-- ctx.corpus.symbols is a regular Lua sequence: 1-indexed, with `#` and
-- `ipairs`/`pairs` support.

mrdocs.register_transform(function(ctx)
    for _, sym in ipairs(ctx.corpus.symbols) do
        if sym.kind == "function" then
            sym.name = "renamed_" .. sym.name
            sym.doc = {
                brief = {
                    children = {
                        { kind = "text",
                          literal = "Brief rewritten by Lua extension" }
                    }
                }
            }
        end
    end
end)

File: tests/golden/fixtures/extensions/js-set-name/addons/extensions/rename.js
// Mutate every function from a JavaScript extension. Exercises two
// shapes of direct assignment: a scalar string write (`name`) and a
// nested-object write whose leaves are `Polymorphic<Inline>` values
// selected by a kebab-case `kind` tag.
//
// Mirrors the lua-set-name fixture but exercises the JS path: the
// proxy's `set` trap forwards each assignment into the live C++
// Symbol via reflection.

mrdocs.register_transform("set-name", function(ctx)
{
    for (var i = 0; i < ctx.corpus.symbols.length; ++i)
    {
        var sym = ctx.corpus.symbols[i];
        if (sym.kind === "function")
        {
            sym.name = "renamed_" + sym.name;
            sym.doc = {
                brief: {
                    children: [
                        { kind: "text",
                          literal: "Brief rewritten by JS extension" }
                    ]
                }
            };
        }
    }
});
